handler/perm_handlers_test.go

```diff
@@ -15,21 +15,41 @@ import (
 )
 
 func TestHandler_getPermsByRoleId(t *testing.T) {
-	type mockBehavior func(s *mock_service.MockRolePerm, id int)
+	type mockBehaviorCheck func(s *mock_service.MockAuthUser, perms, role string)
+	type mockBehaviorParseToken func(s *mock_service.MockAuthUser, token string)
+	type mockBehavior func(s *mock_service.MockAuthUser, id int)
 
 	testTable := []struct {
-		name                string
-		input               string
-		id                  int
-		mockBehavior        mockBehavior
-		expectedStatusCode  int
-		expectedRequestBody string
+		name                   string
+		input                  string
+		id                     int
+		inputPerms             string
+		inputRole              string
+		inputToken             string
+		mockBehaviorParseToken mockBehaviorParseToken
+		mockBehavior           mockBehavior
+		mockBehaviorCheck      mockBehaviorCheck
+		expectedStatusCode     int
+		expectedRequestBody    string
 	}{
 		{
-			name:  "OK",
-			input: "1",
-			id:    1,
-			mockBehavior: func(s *mock_service.MockRolePerm, id int) {
+			name:       "OK",
+			input:      "1",
+			id:         1,
+			inputPerms: "",
+			inputRole:  "Superadmin",
+			inputToken: "testToken",
+			mockBehaviorParseToken: func(s *mock_service.MockAuthUser, token string) {
+				s.EXPECT().ParseToken(token).Return(&authProto.UserRole{
+					UserId:      1,
+					Role:        "Superadmin",
+					Permissions: "",
+				}, nil)
+			},
+			mockBehaviorCheck: func(s *mock_service.MockAuthUser, perms, role string) {
+				s.EXPECT().CheckRoleRights(nil, "Superadmin", perms, role).Return(nil)
+			},
```

Comment:
Could you just generate token with correct roles instead of mocking its parsing and check? The less you mock - the more reliable are tests.